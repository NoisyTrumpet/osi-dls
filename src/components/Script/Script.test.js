import React from 'react';
import { render } from '@testing-library/react';
import {
  toBeInTheDocument,
  toHaveAttribute,
  toHaveTextContent
} from '@testing-library/jest-dom/matchers';
import { Script } from './';

const resourceUrl1 = "https://service.force.com/embeddedservice/5.0/esw.min.js";

const testInlineScript = "alert('Inline Script as Text')";

expect.extend({
  toBeInTheDocument,
  toHaveAttribute,
  toHaveTextContent
});

describe('<Script />', () => {
  test('renders', () => {
    const { container } = render(<Script />);
    expect(container.firstChild).toMatchSnapshot();
  });

  test('renders properly using the url prop', () => {
    render(<Script url={resourceUrl1} domId="example" />);

    expect(document.getElementById('example')).toHaveAttribute('src');
  });

  test('renders properly using the text prop', () => {
    render(<Script text={testInlineScript} domId="example2" />);

    expect(document.getElementById('example2')).toHaveTextContent(testInlineScript);
  });
});
