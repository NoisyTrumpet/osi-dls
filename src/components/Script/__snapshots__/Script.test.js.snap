// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Script /> renders 1`] = `null`;
